torch/fx/graph_module.py

```diff
@@ -642,12 +642,50 @@ def code(self) -> str:
         Return the Python code generated from the ``Graph`` underlying this
         ``GraphModule``.
         """
+
+        self.real_recompile()
         if not hasattr(self, '_code'):
             raise RuntimeError('Code has not been generated! Please report a bug to PyTorch')
         return self._code
 
+    @property
+    def in_spec(self):
+        self.real_recompile()
+        return getattr(self, "_in_spec", None)
+
+    @property
+    def out_spec(self):
+        self.real_recompile()
+        return getattr(self, "_out_spec", None)
```

Comment:
This feels extra awkward. It seems like you have a BC-breaking change (previous access to `_out_spec` is now `out_spec`) PLUS you won't see anything is wrong until it breaks! If you *must* break it, best to rename the old underscored fields to something else!

Reply:
I totally agree :) The reason that I do this BC breaking change is because I think
1. '_in_spec' / '_out_spec' is some private fields that normal user is not supposed to use
2. there are only very small number of usages of them 

I can keep BC compatible by changing `_in_spec` into a property and let it access an attributed called something like `__in_spec` (double underscore) . But an alternative is we just create an property called `in_spec` and change the places previously accessing `_in_spec` to access `in_spec` instead. My assumption is `_in_spec`/`_out_spec` does not belong to public APIs, so doing some BC breaking change should be fine. Let me know if I'm wrong. I'll be glad to change my implementation

Reply:
I think `renaming` is a good idea. I can do something like:
1. create a field `_graph_in_spec`/`_graph_out_spec` that serves the same purpose as what `_in_spec`/`_out_spec` is doing
2. create property `graph_in_spec`/`graph_out_spec` to access _graph_in_spec and _graph_out_spec and do recompilation as necessary
3. change `_in_spec`/`_out_spec` to alias to `graph_in_spec`/`graph_out_spec` and deprecate their usage. (Q: what API should I use to deprecate a property?)